### PR TITLE
Update Python dependencies

### PR DESCRIPTION
**Pull request by jonatanschroeder**
In the same spirit as #8402, updating Python dependencies before a flurry of dependabot PRs comes in.

### CODE REVIEW

File: images/plbase/python-requirements.txt [hunk 1]
```diff
@@ -15,33 +15,33 @@ networkx==3.1
 nltk==3.8.1
 numpy==1.25.2
 openpyxl==3.1.2
-pandas-stubs==2.0.2.230605
-pandas==2.0.3
+pandas-stubs==2.0.3.230814
+pandas==2.1.0
```
Comment (eliotwrobson) [inline images/plbase/python-requirements.txt:19]: Should stick to pandas 2.0.3 for now if you want to be safe. We want `pandas` to match the first 3 numbers in the `pandas-stubs` version.

File: images/plbase/python-requirements.txt [hunk 2]
```diff
@@ -15,33 +15,33 @@ networkx==3.1
...
 Pint==0.22
 psutil==5.9.5
 PuLP==2.7.0
 pycryptodome==3.18.0
 Pygments==2.16.1
 pygments-ansi-color==0.3.0
-pygraphviz==1.6
+pygraphviz==1.11
```
Comment (eliotwrobson) [inline images/plbase/python-requirements.txt:26]: @nwalters512 I think there was some issues with upgrading this?

File: graders/python/requirements.txt [hunk 3]
```diff
@@ -1,25 +1,25 @@
-automata-lib==7.1.0
+automata-lib==8.0.0
```
Comment (eliotwrobson) [inline graders/python/requirements.txt:1]: Discussed with @nwalters512, we think we can actually remove `automata-lib` from the grader image.

**Review by eliotwrobson — CHANGES_REQUESTED** [review]
Some discussion items to resolve. The rest of the version changes seem reasonable.

File: images/plbase/python-requirements.txt (hunk 1 above)
Comment (jonatanschroeder) [reply images/plbase/python-requirements.txt]: Done in 092e1d89b.

File: graders/python/requirements.txt (hunk 3 above)
Comment (jonatanschroeder) [reply graders/python/requirements.txt]: Restored to 7.1.0. I'll leave the discussion on removing it to a different PR, possibly #8223.

File: images/plbase/python-requirements.txt [hunk 4]
```diff
@@ -31,17 +31,17 @@ pytest-repeat==0.9.1
 pytest==7.4.0
 recommonmark==0.7.1
 regex==2023.8.8
-rpy2==3.5.12
+rpy2==3.5.13
```
Comment (jonatanschroeder) [inline images/plbase/python-requirements.txt:34]:
```suggestion
rpy2==3.5.12
```
3.5.13 is failing to build, both in CI and locally. Restoring as 3.5.12 builds successfully.

**Review by eliotwrobson — APPROVED** [review]

**Pull request merged**